pkg/testdatagen/scenario/e2ebasic.go

```
@@ -578,6 +578,127 @@ func (e e2eBasicScenario) Run(db *pop.Connection, userUploader *uploader.UserUpl
+		},
+	})
+
+	smIDCombo := "f6bd793f-7042-4523-aa30-34946e7339c9"
+	smWithCombo := testdatagen.MakeExtendedServiceMember(db, testdatagen.Assertions{
+		ServiceMember: models.ServiceMember{
+			ID:            uuid.FromStringOrNil(smIDCombo),
+			UserID:        uuid.FromStringOrNil(uuidStr),
+			FirstName:     models.StringPointer("Submitted"),
+			LastName:      models.StringPointer("Ppmhhg"),
+			Edipi:         models.StringPointer("6833908165"),
+			PersonalEmail: models.StringPointer(email),
+		},
+	})
+	// currently don't have "combo move" selection option, so testing ppm office when type is HHG
+	selectedMoveType := models.SelectedMoveTypeHHG
+	move := testdatagen.MakeMove(db, testdatagen.Assertions{
+		Order: models.Order{
+			ServiceMemberID: uuid.FromStringOrNil(smIDCombo),
+			ServiceMember:   smWithCombo,
+		},
+		Move: models.Move{
+			ID:               uuid.FromStringOrNil("7024c8c5-52ca-4639-bf69-dd8238308c98"),
+			Locator:          "COMBOS",
+			SelectedMoveType: &selectedMoveType,
+		},
+	})
+
+	estimatedHHGWeight := unit.Pound(1400)
+	actualHHGWeight := unit.Pound(2000)
+	testdatagen.MakeMTOShipment(db, testdatagen.Assertions{
+		MTOShipment: models.MTOShipment{
+			ID:                   uuid.FromStringOrNil("8689afc7-84d6-4c60-a739-8cf96ede2606"),
+			PrimeEstimatedWeight: &estimatedHHGWeight,
+			PrimeActualWeight:    &actualHHGWeight,
+			ShipmentType:         models.MTOShipmentTypeHHG,
+			ApprovedDate:         swag.Time(time.Now()),
+			Status:               models.MTOShipmentStatusSubmitted,
+			MoveTaskOrder:        move,
+			MoveTaskOrderID:      move.ID,
+		},
+	})
+
+	testdatagen.MakePPM(db, testdatagen.Assertions{
+		ServiceMember: move.Orders.ServiceMember,
+		PersonallyProcuredMove: models.PersonallyProcuredMove{
+			OriginalMoveDate: &nextValidMoveDate,
+			Move:             move,
+			MoveID:           move.ID,
+		},
+		UserUploader: userUploader,
+	})
+
+	move.Submit(time.Now())
+	models.SaveMoveDependencies(db, &move)
```

I still see the PPM status set to DRAFT in the move list but wondering why this doesn't change it to SUBMITTED.  Is there a silent error happening somewhere?

---

It looks like the move doesn't have any ppm associations because it was created after the local var so the status isn't updated.  Maybe refetching the move or manually adding the associated ppm before calling submit and save will do the trick?

---

ooh, good catch!  I'll work on that.